src/syscall/syscall_libc.go

```diff
@@ -1,26 +1,27 @@
-// +build darwin
+// +build darwin nintendoswitch
 
 package syscall
 
 import (
+	"errors"
 	"unsafe"
 )
 
 func Close(fd int) (err error) {
-	return ENOSYS // TODO
+	return ENOSYS
 }
 
 func Write(fd int, p []byte) (n int, err error) {
 	buf, count := splitSlice(p)
 	n = libc_write(int32(fd), buf, uint(count))
 	if n < 0 {
-		err = getErrno()
+		err = errors.New("error writing to fd") // TODO: Return errno
```

Comment:
I would prefer if you defined `getErrno` so that errno remains correct for other platforms (for example macOS). If there is no way to get the errno currently, you can implement a dummy `getErrno` (in syscall_nintendoswitch.go).

```go
var dummyError = errors.New("unknown syscall error")
func getErrno() error {
    return dummyError
}
```

Reply:
LIkewise @racerxdl this comment too.

Reply:
Done!